packages/svelte/src/reactivity/media-query.js

```diff
@@ -0,0 +1,35 @@
+import { get } from '../internal/client/runtime.js';
+import { set, source } from '../internal/client/reactivity/sources.js';
+import { effect_tracking } from '../internal/client/reactivity/effects.js';
+import { createStartStopNotifier } from './start-stop-notifier.js';
+
+/**
+ * Creates a media query and provides a `current` property that reflects whether or not it matches.
+ */
+export class MediaQuery {
+	#version = source(0);
+	#query;
+	#notify;
```

Comment:
we could do the setup here instead of in the constructor, feels slightly neater. we can also use `on` (from `../events/index.js`) and `increment` (from `./utils.js`)

```suggestion
	#notify = createStartStopNotifier(() => {
		return on(this.#query, 'change', () => increment(this.#version));
	});
```